### PR TITLE
Muon o sselection

### PR DESCRIPTION
**Pull request by amartelli**
I just realised that the OS request was there only for electrons. I think that for AccXEff it's better to introduce this in the analyser, because a cut a posteriori can give to an event rejected, while in the analyser a different lepton-lepton combination can be considered

### CODE REVIEW

**Comment by tstreble** [comment]
Ok the change in OS for mumu is fine.
This includes also #19 though.
Closing #19 and moving the comments here

**Comment by tstreble** [comment]
I'm not sure we really expect a big difference in acceptance x efficiency vs pT x eta for the different q2 bins.
Splitting by q2 will just reduce the stat I think.
What may be useful however are the 2D gen distributions at the muon tag level, which will be impacted by the q2 requirement.

File: NtupleProducer/macro/computeAE_charged_plotEff.C [hunk 1]
```diff
@@ -0,0 +1,471 @@
...
+#include "TMath.h"
+#include "TApplication.h"
+#include "TError.h"
+#include "TCanvas.h"
+#include "TGraphErrors.h"
+#include "TPad.h"
+#include "TMultiGraph.h"
+#include "TStyle.h"
+#include "TChain.h"
+
```
Comment (tstreble) [inline NtupleProducer/macro/computeAE_charged_plotEff.C]: TLegend.h is missing

File: NtupleProducer/macro/computeAE_charged_plotEff.C [hunk 2]
```diff
@@ -0,0 +1,471 @@
...
+  //selections
+  std::string cut_muonTag = "Muon_sel_index != -1";
+  // should require that muonTag does not match with gen probes?
+  std::string cut_genAcc = "GenPart_e2FromB_index != -1 && GenPart_pt[GenPart_e2FromB_index] > 1. && abs(GenPart_eta[GenPart_e2FromB_index]) < 2.4 && GenPart_e1FromB_index != -1 && GenPart_pt[GenPart_e1FromB_index] > 1. && abs(GenPart_eta[GenPart_e1FromB_index]) < 2.4 && GenPart_KFromB_index != -1 && GenPart_pt[GenPart_KFromB_index] > 1. && abs(GenPart_eta[GenPart_KFromB_index]) < 2.4";
+  std::string cut_genEff = "BToKee_gen_index != -1 && BToKee_gen_index == BToKee_sel_index";
+  std::string cut_chargeEff = "BToKee_ele1_charge[BToKee_gen_index]*BToKee_ele2_charge[BToKee_gen_index] < 0.";
+  std::string cut_alphaEff = "BToKee_cosAlpha[BToKee_gen_index] > 0.99";
+  std::string cut_vtxCLEff = "BToKee_CL_vtx[BToKee_gen_index] > 0.1";  
+  std::string cut_DCAEff  = "abs(BToKee_kaon_DCASig[BToKee_gen_index]) > 6";
```
Comment (tstreble) [inline NtupleProducer/macro/computeAE_charged_plotEff.C]: The cut should rather be on Lxy if the goal is to replicate Riccardo's selections.

Comment (tstreble) [reply NtupleProducer/macro/computeAE_charged_plotEff.C]: The fact that it does nothing to the signal is good news :)
DCASig>6 however is already reducing significantly the signal.
The 3.3 currently applied is inherited from BPH-15-001 http://cms.cern.ch/iCMS/jsp/db_notes/noteInfo.jsp?cmsnoteid=CMS%20AN-2014/244 (section 3.1.4)
but it already seems to be tight for the signal.
This is something which has not yet been optimised and I would be reluctant to arbitrarily go to DCASig>6, without checking the benefit in terms of background discrimination, as it cuts further in the bulk of the signal.
In terms of background discrimination, I would expect something similar to what was seen for the B->Kpipi final state
https://indico.cern.ch/event/739836/contributions/3053808/attachments/1676756/2692580/18.06.28_First_look_at_BToKpipi_final_state.pdf (slide 12, top right)

**Comment by amartelli** [comment]
I should have implemented all the fixes, I fully agree with all your comments

**Pull request merged**